src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp

```diff
@@ -529,4 +531,107 @@ void SetExeNetworkInfo(const std::shared_ptr<IExecutableNetworkInternal>& exeNet
+    m_plugin_name = m_old_plugin->GetName();
+    m_is_new_api = m_old_plugin->IsNewAPI();
+    m_core = m_old_plugin->GetCore();
+    m_executor_manager = m_old_plugin->executorManager();
+}
+
+std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> IPluginWrapper::compile_model(
+    const std::shared_ptr<const ov::Model>& model,
+    const ov::AnyMap& properties) const {
+    auto exec_network =
+        m_old_plugin->LoadNetwork(ov::legacy_convert::convert_model(model, is_new_api()), any_copy(properties));
+    return exec_network;
+}
+
+std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> IPluginWrapper::compile_model(
+    const std::string& model_path,
+    const ov::AnyMap& properties) const {
+    auto exec_network = m_old_plugin->LoadNetwork(model_path, any_copy(properties));
+    return exec_network._ptr;
+}
+
+std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> IPluginWrapper::compile_model(
+    const std::shared_ptr<const ov::Model>& model,
+    const ov::AnyMap& properties,
+    const ov::RemoteContext& context) const {
+    auto compiled_model = m_old_plugin->LoadNetwork(ov::legacy_convert::convert_model(model, is_new_api()),
+                                                    any_copy(properties),
+                                                    context._impl);
+    return compiled_model;
+}
+
+void IPluginWrapper::set_property(const ov::AnyMap& properties) {
+    m_old_plugin->SetProperties(properties);
+}
+
+ov::Any IPluginWrapper::get_property(const std::string& name, const ov::AnyMap& arguments) const {
+    try {
+        return m_old_plugin->GetConfig(name, arguments);
+    } catch (...) {
+        return m_old_plugin->GetMetric(name, arguments);
+    }
+}
+
+ov::RemoteContext IPluginWrapper::create_context(const ov::AnyMap& remote_properties) const {
+    return ov::RemoteContext{m_old_plugin->CreateContext(remote_properties), {nullptr}};
+}
+
+ov::RemoteContext IPluginWrapper::get_default_context(const ov::AnyMap& remote_properties) const {
+    return ov::RemoteContext{m_old_plugin->GetDefaultContext(remote_properties), {nullptr}};
+}
+
+std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> IPluginWrapper::import_model(
+    std::istream& model,
+    const ov::AnyMap& properties) const {
+    return m_old_plugin->ImportNetwork(model, any_copy(properties));
+}
+
+std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> IPluginWrapper::import_model(
+    std::istream& model,
+    const ov::RemoteContext& context,
+    const ov::AnyMap& properties) const {
+    return m_old_plugin->ImportNetwork(model, context._impl, any_copy(properties));
+}
+
+ov::SupportedOpsMap IPluginWrapper::query_model(const std::shared_ptr<const ov::Model>& model,
+                                                const ov::AnyMap& properties) const {
+    auto res = m_old_plugin->QueryNetwork(ov::legacy_convert::convert_model(model, is_new_api()), any_copy(properties));
```

Comment:
will it be a problem here because of preprocessing? E.g. some info is missed or extra operations are reported

Reply:
I don't think so. Our preprocessing contains a very limited set of operations which should be supported by all plugins

Reply:
Here we don't wrap legacy pre-processing to the model, we restore it from the rt info. So it means that CNNNetwork will be the same as it was in the original API.

Reply:
Disscused and decided to keep as is in order to don't expose new operations and new names from legacy `QueryNetwork`